probe-rs/src/debug/variable.rs

```diff
@@ -655,214 +699,250 @@ impl Variable {
+    indentation: usize,
+    children: &mut (impl Iterator<Item = &'a Variable> + Clone),
+    type_name: &str,
+) -> String {
+    let line_start = line_indent_string(indentation);
+
+    // Limit arrays to 10 elements
+    const ARRAY_MAX_LENGTH: usize = 10;
+
+    // If we at least ARRAY_MAX_LENGTH + 2 items in the iterator, cap at ARRAY_MAX_LENGTH.
+    // If we have less, cap at the actual number of items.
+    // This helps us to never write "and 1 more" with the reasoning that the space used for this
+    // text, can be used for printing that one item.
+    let count = children.clone().count();
+    let take = if count > ARRAY_MAX_LENGTH + 1 {
+        ARRAY_MAX_LENGTH
+    } else {
+        count
+    };
+
+    let children_values = children
+        .by_ref()
+        .take(take)
+        .filter_map(|child| child.formatted_variable_value(variable_cache, indentation + 1, false))
+        .join(",");
+
+    let remainder = if count > ARRAY_MAX_LENGTH + 1 {
+        format!(",\n{line_start}\t... and {} more", count - take)
+    } else {
+        String::new()
+    };
+
+    format!("{line_start}{type_name} = [{children_values}{remainder}{line_start}]")
+}
+
+/// Format any struct like value .
+///
+/// Recursively formats all child values.
+fn format_struct_value<'a>(
+    variable_cache: &VariableCache,
+    indentation: usize,
+    children: &mut (impl Iterator<Item = &'a Variable> + Clone),
+    type_name: &str,
+) -> String {
+    let line_start = line_indent_string(indentation);
+
+    // FIXME: this is not hit by any of the unwind tests, which is weird because
+    // some of them contain `Some` structs.
+    // Handle special structure types like the variant values of `Option<>` and `Result<>`
+    let children_values = format_children_values(variable_cache, indentation, children, false);
+
+    format!("{line_start}{type_name} = ({children_values})")
+}
+
+/// Format any root value.
+///
+/// Recursively formats all child values.
+fn format_root_value<'a>(
+    variable_cache: &VariableCache,
+    indentation: usize,
+    children: &mut (impl Iterator<Item = &'a Variable> + Clone),
+    type_name: &str,
+) -> String {
+    let line_start = line_indent_string(indentation);
+
+    let children_values = format_children_values(variable_cache, indentation, children, true);
+    format!("{line_start}{type_name} {{{children_values}{line_start}}}")
+}
+
+/// Format any value that has no type that requires special handling.
+///
+/// Recursively formats all child values.
+fn format_default_value<'a>(
+    variable_cache: &VariableCache,
+    indentation: usize,
+    name: &VariableName,
+    children: &mut (impl Iterator<Item = &'a Variable> + Clone),
+    type_name: &String,
+    show_name: bool,
+) -> String {
+    let line_start = line_indent_string(indentation);
+
+    // Find the first child of the structure if it exists.
+    let child = children
+        .clone()
+        .find(|c| matches!(&c.name, VariableName::Named(_)));
```

Comment:
We should maybe introduce `is_named()` on the enum. Or on Variable for that matter, right now it reads like "Find the first child of the structure whose name is named" 😅